tico/serialize/pack.py

```diff
@@ -0,0 +1,34 @@
+# Copyright (c) 2025 Samsung Electronics Co., Ltd. All Rights Reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+
+
+def pack_buffer(flat_data: np.ndarray, dtype: str) -> np.ndarray:
+    assert len(flat_data.shape) == 1
+
+    if dtype == "uint4":
+        if flat_data.dtype != np.uint8:
+            raise RuntimeError("uint4 data should be saved in uint8.")
+
+        numel = flat_data.shape[0]
+        packed = np.zeros((numel + 1) // 2, dtype=np.uint8)
+        for i in range(numel):
+            if i % 2 == 0:
+                packed[i // 2] = flat_data[i] & 0x0F
+            else:
+                packed[i // 2] |= flat_data[i] << 4
+        return packed
```

Comment:
```suggestion
        for i in range(numel):
            if not ((flat_data[i] >=0) or (flat_data[i] <=255)):
                raise RuntimeError("flat_data[i] is out of range uint4.")
            if i % 2 == 0:
                packed[i // 2] = flat_data[i] & 0x0F
            else:
                packed[i // 2] |= flat_data[i] << 4
        return packed
```
Is it guaranteed that `flat_data[i]` is bounded in the range of `uint4`?
Is there no possibility like that the `flat_data` is out of `uint4` range though the `dtype` is specified as `uint4`?
IMHO, it would be nice to check to ensure that the `flat_data` is in the range of `uint4`

Reply:
Range of weights values is checked in `infer_dtype`. I'll add an assert rather than exception.